app/src/androidTest/java/com/github/sdp/mediato/ProfileFragmentTest.java

```diff
@@ -111,6 +117,63 @@ public void testAddMediaButton() {
     collectionRecyclerView.check(matches(hasItemCount(initialItemCount + 1)));
   }
 
+  // Test the initial state of the list of collections after profile creation
+  @Test
+  public void testInitialOuterRecyclerViewState() {
+    ViewInteraction outerRecyclerView = onView(withId(R.id.collection_list_recycler_view));
+
+    // Check that the outer RecyclerView is displayed
+    outerRecyclerView.check(matches(isDisplayed()));
+
+    // Check that the outer RecyclerView initially has one item (default collection)
+    outerRecyclerView.check(matches(hasItemCount(1)));
+  }
+
+  // Check that a new collection has been added to the outer RecyclerView if the user chooses a valid collection name
+  @Test
+  public void testAddValidCollection() throws UiObjectNotFoundException {
+    ViewInteraction outerRecyclerView = onView(withId(R.id.collection_list_recycler_view));
+    ViewInteraction addCollectionButton = onView(withId(R.id.add_collection_button));
+
+    int initialItemCount = getRecyclerViewItemCount(R.id.collection_list_recycler_view);
+    outerRecyclerView.check(matches(hasItemCount(initialItemCount)));
+    addCollectionButton.perform(click());
+    enterTextInAlertBox("valid");
+
+    outerRecyclerView.check(matches(hasItemCount(initialItemCount + 1)));
+  }
+
+  // Check that no new collection has been added to the outer RecyclerView if the user enters an empty collection name
+  @Test
+  public void testEmptyCollectionNameNotAdded() throws UiObjectNotFoundException {

```

Comment:
This is valid for all the tests: Are you updating the collections in the database in the code that you're testing here ? If so, you could use the emulator in this class like in the DatabaseTests classes.

Reply:
The database functions are already tested in another class but we should still use the emulator to avoid spamming the "production" database with test values.

Reply:
If I just start the database emulator before testing it won't upload to the actual database? Even if it is not in a database test?

Reply:
Okay thanks, done!